File: Actividad5/Ejercicio1/CarbonFootprintProject/src/model/Bicycle.java
package model;

public class Bicycle implements CarbonFootprint {

    private double distance; // km

    public Bicycle(double distance) {
        this.distance = distance;
    }

    @Override
    public double getCarbonFootprint() {

        if (distance < 0) {
            return 0;
        }

        return distance * 0.01;
    }

    @Override
    public String toString() {
        return "Bicycle | Distance: " + distance;
    }
}

File: Actividad6/Ejercicio 1/CarbonFootprintPy/model/Bicycle.py
from abc import ABC, abstractmethod

from model import CarbonFootprint

class Bicycle(CarbonFootprint):
    def __init__(self, distance: float):
        self.distance = distance if distance >= 0 else 0

    def get_carbon_footprint(self) -> float:
        return self.distance * 0.01

    def __str__(self):
        return f"Bicycle | Distance: {self.distance} km"
    